openapi/src/parts/controlled_gcp_dataproc_cluster.yaml

```diff
@@ -0,0 +1,378 @@
+        jobReport:
+          $ref: '#/components/schemas/JobReport'
+        errorReport:
+          $ref: '#/components/schemas/ErrorReport'
+        dataprocCluster:
+          $ref: '#/components/schemas/GcpDataprocClusterResource'
+
+    GcpDataprocClusterCreationParameters:
+      type: object
+      required:
+      - clusterId
+      - region
+      - configBucket
+      - tempBucket
+      - managerNodeConfig
+      - primaryWorkerConfig
+      properties:
+        clusterId:
+          type: string
+          description: Dataproc cluster name.
+        region:
+          type: string
+          description: GCP region.
+        initializationScript:
+          type: string
+          description: Path to the initialization script.
+        components:
+          type: array
+          items:
+              type: string
+          uniqueItems: true
+          description: List of optional components to install on the cluster manager node. See full list of available components here https://cloud.google.com/dataproc/docs/concepts/components/overview
+        softwareFramework:
+          type: string
+          enum: ["HAIL"]
+          description: Software framework to setup the cluster with. Currently supported values are ['hail'].
+        configBucket:
+          type: string
+          format: uuid
+          description: Staging bucket controlled resource uuid.
+        tempBucket:
+          type: string
+          format: uuid
+          description: Temp bucket controlled resource uuid.
+        autoscalingPolicy:
+          type: string
+          description: URI of autoscaling policy.
+        metadata:
+          type: object
+          description: Metadata key-value pairs.
+          additionalProperties:
+            type: string
+        managerNodeConfig:
+          $ref: '#/components/schemas/GcpDataprocClusterInstanceGroupConfig'
+        primaryWorkerConfig:
+          $ref: '#/components/schemas/GcpDataprocClusterInstanceGroupConfig'
+        secondaryWorkerConfig:
+          $ref: '#/components/schemas/GcpDataprocClusterInstanceGroupConfig'
+        lifecycleConfig:
+          $ref: '#/components/schemas/GcpDataprocClusterLifecycleConfig'
+
+    GcpDataprocClusterInstanceGroupConfig:
+      type: object
+      required:
+      - numInstances
+      - machineType
+      properties:
+        numInstances:
+          type: integer
+          format: int32
+          description: Number of instances in the instance group.
+        machineType:
+          type: string
+          description: Compute engine machine type.
+        imageUri:
+          type: string
+          description: Compute engine image used for cluster instances.
+        acceleratorConfig:
+          $ref: '#/components/schemas/GcpDataprocClusterAcceleratorConfig'
+        diskConfig:
+          $ref: '#/components/schemas/GcpDataprocClusterDiskConfig'
+        preemptibility:
+          type: string
+          enum: [ "SPOT", "PREEMPTIBLE" ]
```

Comment:
what does SPOT means?

Reply:
https://cloud.google.com/dataproc/docs/concepts/compute/secondary-vms#preemptible_and_non-preemptible_secondary_workers.

Adding ^ to description of `preemptibility`